### ISSUE TITLE
The no-mergeable-selectors rule makes a bad suggestion

### ISSUE THREAD

**Issue by chrismbarr** [sasstools/sass-lint#858]
I am using the current version of `gulp-sass-lint`

I have this code 
```scss
.is-sticky {
	//Add this class when an element is stuck to the page
	&:not(.no-sticky-shadow) {
		//A class could be added to prevent this shadow from being added
		box-shadow: $shadow-z-2;
	}

	&:not(.card) {
		//Only add these for non-cards
		background-color: $card-bg;
		padding: $spacing $spacing-small 0 $spacing-small;
	}
}
```

Which produces this error
```
Rule `.is-sticky &:not(null)` should be merged with the rule on line 152
```

I don't understand why or how those rules could be merged, and it's apparently not even parsing these rules correctly due to the `null` being in there.  My guess it that it parses each of these rules as `:not(null)` which makes it appear they are the same selector and could be merged.

**Comment by DanPurdy** [comment]
Yeah, it's a bug. I'll look into it, it's not reading the content inside the parenthesis correctly. I thought i'd caught all cases of this but obviously not. Sorry!

**Comment by chrismbarr** [comment]
In the meantime I was able to add this to my configuration to ignore this false positive

```yml
no-mergeable-selectors:
    - 1
    -
      'whitelist':
        - '.is-sticky &:not(null)'
```

**Comment by DanPurdy** [comment]
Hi @chrismbarr can you just check which version of sass-lint you're using?

This seems to be working correctly in 1.9.1

See below screenshots
![screen shot 2016-09-01 at 23 10 06](https://cloud.githubusercontent.com/assets/1386054/18186184/512e5eee-7099-11e6-8774-5a8c92e2ed1e.png)
Your example passing

![screen shot 2016-09-01 at 23 10 18](https://cloud.githubusercontent.com/assets/1386054/18186191/5aaa2566-7099-11e6-81ab-c31393abb8dd.png)
failing as it should if i make them the same.

**Comment by chrismbarr** [comment]
Ah, you're right.  sorry, my fault.  It looks like `gulp-sass-lint` only pulled down `sass-lint` version `1.9.0` initially.  I've fix this to pull down `1.9.1` now and it does work correctly.
